crates/oxc_span/src/source_type/mod.rs

```diff
@@ -269,68 +265,50 @@ impl SourceType {
     }
 
     #[must_use]
-    pub const fn with_script(mut self, yes: bool) -> Self {
-        if yes {
-            self.module_kind = ModuleKind::Script;
-        }
+    pub const fn set_script(mut self) -> Self {
+        self.module_kind = ModuleKind::Script;
         self
     }
 
     #[must_use]
-    pub const fn with_module(mut self, yes: bool) -> Self {
-        if yes {
-            self.module_kind = ModuleKind::Module;
-        } else {
-            self.module_kind = ModuleKind::Script;
-        }
+    pub const fn set_module(mut self) -> Self {
+        self.module_kind = ModuleKind::Module;
         self
     }
 
     #[must_use]
-    pub const fn with_unambiguous(mut self, yes: bool) -> Self {
-        if yes {
-            self.module_kind = ModuleKind::Unambiguous;
-        }
+    pub const fn set_unambiguous(mut self) -> Self {
+        self.module_kind = ModuleKind::Unambiguous;
         self
     }
 
     #[must_use]
-    pub const fn with_javascript(mut self, yes: bool) -> Self {
-        if yes {
-            self.language = Language::JavaScript;
-        }
+    pub const fn set_js(mut self) -> Self {
```

Comment:
praise: I much prefer the intent being in the name over passing a Boolean parameter